src/Diagnostics.DataProviders/DataProviders/MdmDataProvider.cs

```diff
@@ -206,5 +224,71 @@ public async Task<IEnumerable<DataTable>> GetMultipleTimeSeriesAsync(DateTime st
 
             return result;
         }
+
+        private void AddMdmInformationToMetadata(IEnumerable<Tuple<string, string, IEnumerable<KeyValuePair<string, string>>>> definitions, 
+            DateTime startTimeUtc, 
+            DateTime endTimeUtc, 
+            int seriesResolutionInMinutes)
+        {
+
+            var queryParameters = new List<MdmQueryParameters>();
+            var dashboard = string.Empty;
+
+            foreach (var d in definitions)
+            {
+                //Item1 - NameSpace
+                //Item2 - MetricName
+
+                if (string.IsNullOrWhiteSpace(dashboard))
+                {
+                    if (d.Item1 == "Microsoft/Web/AppServicePlans")
+                    {
+                        dashboard = "WAWS Shoebox/App Service Plans/Per Resource Instance";
+                    }
+                    else if (d.Item1 == "Microsoft/Web/WebApps")
+                    {
+                        dashboard = "WAWS Shoebox/Web Apps/Per Resource";
+                    }
+                }
+
+                foreach (var instanceDimension in d.Item3.Where(x => x.Key != "ServerName"))
+                {
+
+                    var queryValue = $"//*[id='{instanceDimension.Key}']";
+                    if (!queryParameters.Any(x => x.query == queryValue))
+                    {
+                        var mdmParameter = new MdmQueryParameters
+                        {
+                            query = queryValue,
+                            replacement = instanceDimension.Value
+                        };
+
+                        queryParameters.Add(mdmParameter);
+                    }
+                }
+            }
+
+            if (queryParameters.Count > 0 && !string.IsNullOrWhiteSpace(dashboard))
+            {
+                bool urlExists = false;
+                string finalUrl = string.Empty;
+                var urlParmeters = $"dashboard/{dashboard}?overrides={JsonConvert.SerializeObject(queryParameters)}&globalStartTime={GetDateTimeInEpochMilliseconds(startTimeUtc)}&globalEndTime={GetDateTimeInEpochMilliseconds(endTimeUtc)}&pinGlobalTimeRange=true";
+
+                finalUrl = $"https://jarvis-west.dc.ad.msft.net/{urlParmeters} ";
+
+                urlExists = Metadata.PropertyBag.Any(x => x.Key == "Query" &&
+                                                            x.Value.GetType() == typeof(DataProviderMetadataQuery) &&
+                                                            x.Value.CastTo<DataProviderMetadataQuery>().Url.Equals(finalUrl, StringComparison.OrdinalIgnoreCase));
+                if (!urlExists)
+                {
+                    Metadata.PropertyBag.Add(new KeyValuePair<string, object>("Query", new DataProviderMetadataQuery() { Text = finalUrl, Url = finalUrl }));
+                }
+            }
+        }
+
+        private double GetDateTimeInEpochMilliseconds(DateTime dateTimeUtc)
```

Comment:
You can use `DateTimeOffset` method of `ToUnixTimeMilliseconds`. https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.tounixtimemilliseconds?view=netframework-4.8

Reply:
Sweet! I didn't know this existed